### PR TITLE
chore(webchat): store and load user lang in browser storage

### PR DESCRIPTION
**Pull request by laurentlp**
This PR adds the storage class (copied from ui-lite) and uses it to store and fetch the user language in the browser's storage (local, session, cookie).

### CODE REVIEW

File: packages/webchat/src/core/api.tsx [hunk 1]
```diff
@@ -13,12 +13,14 @@ export default class WebchatApi {
 
   constructor(private socket: BpSocket) {}
 
+  // TODO: Do we still need this?
   private get baseUserPayload() {
     return {
       webSessionId: window.__BP_VISITOR_SOCKET_ID
     }
   }
```
Comment (samuelmasse) [inline packages/webchat/src/core/api.tsx:21]: We don't need this no we can delete it.

File: packages/webchat/src/core/api.tsx [hunk 2]
```diff
@@ -13,12 +13,14 @@ export default class WebchatApi {
...
 
+  // TODO: Fix this
   async fetchBotInfo() {
     try {
       const { data } = await this.axios.get('/botInfo', this.axiosConfig)
```
Comment (samuelmasse) [inline packages/webchat/src/core/api.tsx:26]: This one will be done by fetching the botInfo.json file that will be located at https://mediafiles.botpress.dev/clientId/info.json

File: packages/webchat/src/core/api.tsx [hunk 3]
```diff
@@ -78,6 +63,7 @@ export default class WebchatApi {
     }
   }
 
+  // TODO: Fis this
   async resetSession(conversationId: uuid): Promise<void> {
     try {
       await this.axios.post('/conversations/reset', { ...this.baseUserPayload, conversationId }, this.axiosConfig)
```
Comment (samuelmasse) [inline packages/webchat/src/core/api.tsx:69]: Maybe this should be considered debugger stuff? If a user wants a new conversation we can simply make an API call to create a new conversation. There is no need for a "reset conversation" concept

File: packages/webchat/src/core/api.tsx [hunk 4]
```diff
@@ -103,6 +89,7 @@ export default class WebchatApi {
     }
   }
 
+  // TODO: Fix this
   async downloadConversation(conversationId: uuid): Promise<any> {
     try {
       const { data } = await this.axios.post(
```
Comment (samuelmasse) [inline packages/webchat/src/core/api.tsx:95]: That's doable in frontend

File: packages/webchat/src/core/api.tsx [hunk 5]
```diff
@@ -135,6 +122,7 @@ export default class WebchatApi {
     }
   }
 
+  // TODO: Fix this
   async deleteMessages(conversationId: uuid) {
     try {
       await this.axios.post(
```
Comment (samuelmasse) [inline packages/webchat/src/core/api.tsx:128]: There's a API route already to delete all messages of a conversation. We could add a way to do this by socket as well.

File: packages/webchat/src/utils/storage.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+let keyPrefix = ''
+const setKeyPrefix = (prefix: string) => {
+  keyPrefix = prefix
+}
+
+const formatKey = (key: string) => {
+  return `${keyPrefix}-${key}`
+}
+
+let useSessionStorage = new Boolean(window.USE_SESSION_STORAGE)
```
Comment (samuelmasse) [inline packages/webchat/src/utils/storage.ts:19]: When is USE_SESSION_STORAGE true?

File: packages/webchat/src/store/composer.ts [hunk 7]
```diff
@@ -52,7 +50,7 @@ class ComposerStore {
       this._sentHistoryIndex = 0
 
       if (this.rootStore.config.enablePersistHistory) {
-        window.BP_STORAGE?.set(SENT_HISTORY_KEY, takeRight(this._sentHistory, constants.SENT_HISTORY_SIZE))
+        window.BP_STORAGE.set(SENT_HISTORY_KEY, takeRight(this._sentHistory, constants.SENT_HISTORY_SIZE))
```
Comment (samuelmasse) [inline packages/webchat/src/store/composer.ts:53]: BP_STORAGE is defined? Where is that created?

**Review by samuelmasse — COMMENTED** [review]
How do we test this? Also it seems like the data about language is stored in a different way than the user creds. They should probably use the same code. I'm also not a fan of having a global window.BP_STORAGE item, just importing functions would be better.

File: packages/webchat/src/core/api.tsx (hunk 3 above)
Comment (laurentlp) [reply packages/webchat/src/core/api.tsx:69]: Right! Not sure if this was the case, but maybe this code should have been removed when I deleted the emulator code?

File: packages/webchat/src/core/api.tsx (hunk 4 above)
Comment (laurentlp) [reply packages/webchat/src/core/api.tsx:95]: Perfect adding to the QA task.

File: packages/webchat/src/store/composer.ts (hunk 7 above)
Comment (laurentlp) [reply packages/webchat/src/store/composer.ts:53]: It's kind of a hack, but I import it in the root index.tsx file. I think we should rethink the storage class and make something cleaner and simpler. Maybe in another PR tho?

**Review by samuelmasse — APPROVED** [review]

**Pull request merged**